src/libraries/System.Private.CoreLib/src/System/IO/Stream.cs

```diff
@@ -757,6 +865,20 @@ protected static void ValidateBufferArguments(byte[] buffer, int offset, int cou
             }
         }
 
+        private static void ValidateReadAtLeastArguments(int bufferLength, int minimumBytes)
+        {
+            if (minimumBytes < 0)
+            {
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.minimumBytes, ExceptionResource.ArgumentOutOfRange_NeedNonNegNum);
+            }
+
+            if (bufferLength < minimumBytes)
+            {
+                // TODO: pick the right exception message here
```

Comment:
This one seems the closest one to what you need:
```cs
case ExceptionResource.ArgumentOutOfRange_SmallCapacity:
                    return SR.ArgumentOutOfRange_SmallCapacity;
```

The string is: `capacity was less than the current size`. But maybe we should add a new one if this is not clear enough. Also, it's weird to see that this message starts with lowercase.